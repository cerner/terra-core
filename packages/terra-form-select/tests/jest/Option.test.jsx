import React from 'react';
import Option from '../../src/shared/_Option';

describe('Option', () => {
  it('should render a default Option', () => {
    const wrapper = shallow(<Option value="value" display="display" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a disabled Option', () => {
    const wrapper = shallow(<Option disabled value="value" display="display" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should return true for isOption', () => {
    const option = <Option value="value" display="display" />;
    expect(option.type.isOption).toBeTruthy();
  });
});
