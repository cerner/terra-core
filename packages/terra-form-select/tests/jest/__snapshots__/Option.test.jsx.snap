// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Option should render a default Option 1`] = `
<li
  aria-disabled={false}
  className="option is-default"
  data-terra-select-option={true}
  disabled={false}
  role="option"
  tabIndex="0"
>
  <span
    className="display"
  >
    display
  </span>
</li>
`;

exports[`Option should render a disabled Option 1`] = `
<li
  aria-disabled={true}
  className="option is-default is-disabled"
  data-terra-select-option={true}
  disabled={true}
  role="option"
  tabIndex="0"
>
  <span
    className="display"
  >
    display
  </span>
</li>
`;
